code/go/0chain.net/smartcontract/storagesc/readpool.go

```diff
@@ -82,325 +93,146 @@ func toJson(val interface{}) string {
 	return string(b)
 }
 
-func (rp *readPool) moveToBlobber(sscKey, allocID, blobID string,
-	sp *stakePool, now common.Timestamp, value currency.Coin,
+func (rp *readPool) moveToBlobber(allocID, blobID string,
+	sp *stakePool, value currency.Coin, isOwner bool,
 	balances cstate.StateContextI) (resp string, err error) {
 
-	var cut = rp.blobberCut(allocID, blobID, now)
+	// all redeems to response at the end
+	var redeems []readPoolRedeem
+	var moved currency.Coin
+	currentBalance := rp.OwnerBalance
+	ownership := "owner"
```

Comment:
The variable `ownership` is only for the error block below. We should move all `ownership` related code into that block.

Reply:
I think that'd result in duplicate code in the next block, ie:

```
if value >= currentBalance {
        ownership := "owner"
        if !owner{
               ownership := "visitor"
        }
	return "", fmt.Errorf("not enough tokens in %s read pool for "+
			"allocation: %s, blobber: %s", ownership, allocID, blobID)
}
```

Reply:
I don't follow. Search in file for `ownership`; It only appears three times, when it is defined and in the error block so there will be no duplication if the definition is moved. 

We should keep declarations in the block they are used.

An alternative would be to use an enumerated type for visitor/owner and just use a String method defined off of the type.